File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(arr) {
    let n = arr.length;
    if (n === 0)
        return 0;

    let longest = 1;
    let st = new Set();
    
    for (let i = 0; i < n; i++) {
        st.add(arr[i]);
    }

    for (let it of st) {
        
        if (!st.has(it - 1)) {
            
            let cnt = 1;
            let x = it;
            while (st.has(x + 1)) {
                x = x + 1;
                cnt = cnt + 1;
            }
            longest = Math.max(longest, cnt);
        }
    }
    return longest;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int n = nums.length;
        if(n==0) return 0;
        for(int i = 0; i < n; i++){
            set.add(nums[i]);
        }
        int longest=1;
        for(int it : set){
            if(!set.contains(it-1)){
                int count=1;
                int x=it;
                while(set.contains(x+1)){
                    count++;
                    x+=1;
                }
                longest=Math.max(count,longest);
            }
        }
        return longest;
    }
}